### PR TITLE
Add oneOfIgnoringCase method into StrProp.StrBuilder

### LINKED ISSUE #610: StrProp.StrBuilder is missing the oneOfIgnoringCase method
This method would be similar in function to `oneOf`, but would ignore case.

This might involve adding a new inner class to `StringValidator`, just like `Equals` - perhaps `EqualsIgnoringCase`.  Or rename both to match the new 'oneOf' scheme.  Or add a ignoreCase switch to the class - Any option is fine.

Note that the existing `StringValidator.Equals` has a bug - See #473

### PR DESCRIPTION
**Pull request by alex-kar**
Fixes #610 

### All Submissions:

* Add oneOfIgnoringCase method into StrProp.StrBuilder.
* Cover with tests.

Have you checked that...
* [x] Your pull request is to the *_main_* branch
* [x] Your code is up to date with the latest code from the *_main_*
* [x] You followed the guidelines in our [Developer Guide](https://sites.google.com/view/andhow/developer)?
* [x] Your code does not decrease test coverage (maybe it improves coverage!!)
* [x] If this is related to an issue, please include a link to the issue with the 'Fixes #XXX' syntax.

### CODE REVIEW

File: andhow-core/src/main/java/org/yarnandtail/andhow/valid/StringValidator.java [hunk 1]
```diff
@@ -6,19 +6,19 @@
...
+ *
  * @author ericeverman
  */
 public class StringValidator {
 
 	/**
 	 * Validate that a string is one from the specified set.
 	 */
-	public static class Equals implements Validator<String> {
+	public static class OneOf implements Validator<String> {
```
Comment (eeverman) [inline andhow-core/src/main/java/org/yarnandtail/andhow/valid/StringValidator.java:17]: Nice rename on this.

File: andhow-core/src/main/java/org/yarnandtail/andhow/property/StrProp.java [hunk 2]
```diff
@@ -117,10 +117,14 @@ public StrBuilder mustEqual(String... values) {
...
 		public StrBuilder oneOf(String... values) {
-			this.validation(new StringValidator.Equals(values));
+			this.validation(new StringValidator.OneOf(values));
 			return this;
 		}
 
+		public StrBuilder oneOfIgnoringCase(String... values) {
+			this.validation(new StringValidator.OneOfIgnoringCase(values));
+			return this;
+		}
```
Comment (eeverman) [inline andhow-core/src/main/java/org/yarnandtail/andhow/property/StrProp.java:127]: This method is missing a test.

File: andhow-core/src/test/java/org/yarnandtail/andhow/valid/StringValidatorTest.java [hunk 3]
```diff
@@ -169,65 +169,128 @@ public void regexGetTheValueMustDescriptionTest() {
...
+	/**
+	 * Test of isSpecificationValid method, of class OneOfIgnoringCase.
+	 */
+	@Test
+	public void oneOfIgnoringCaseIsSpecificationValidTest() {
+		StringValidator.OneOfIgnoringCase instance = new StringValidator.OneOfIgnoringCase((String) null);
+		assertFalse(instance.isSpecificationValid());
+
+		instance = new StringValidator.OneOfIgnoringCase();
+		assertFalse(instance.isSpecificationValid());
```
Comment (eeverman) [inline andhow-core/src/test/java/org/yarnandtail/andhow/valid/StringValidatorTest.java:243]: It looks like this constructor ends up passing a null, making it the same test as on line 239.  I think if this were:
```
instance = new StringValidator.OneOfIgnoringCase(new String[0]);
```
It would pick up one more line of test coverage for the empty case, rather than hitting the null case twice.
Similar for the oneOf tests.  (I'm guessing this from the code coverage report that shows the `isSpecificationValid` method as partially covered)

**Review by eeverman — APPROVED** [review]
Nice addition!  The level of testing here is good, but it would be nice to pick up those missing test lines.

**Review by eeverman — APPROVED** [review]
Look great!

File: andhow-core/src/main/java/org/yarnandtail/andhow/property/StrProp.java (hunk 2 above)
Comment (alex-kar) [reply andhow-core/src/main/java/org/yarnandtail/andhow/property/StrProp.java:127]: @eeverman Can you help me with testing this method?
I checked tests for `oneOf` and not sure what exactly I should test there.

Comment (eeverman) [reply andhow-core/src/main/java/org/yarnandtail/andhow/property/StrProp.java:127]: Currently there are no separate tests for PropertyBuilder vs Properties, so any tests we have for this are in `StrProp`.  Here is one way to test it:
```java
	@Test
	public void oneOfShouldBuildCorrectValidation() {
		StrProp.StrBuilder bldr = new StrProp.StrBuilder();
		bldr.oneOf("A", "B");

		assertTrue(bldr._validators.get(0) instanceof StringValidator.Equals); //now renamed
		StringValidator.Equals eqValid = (StringValidator.Equals)(bldr._validators.get(0));
		assertTrue(eqValid.isValid("A") && eqValid.isValid("B"));
	}
```
The test is a bit indirect, but it does verify that the constructed validator seems to be correct.  Beyond that, the Validator tests the validator itself.

Comment (alex-kar) [reply andhow-core/src/main/java/org/yarnandtail/andhow/property/StrProp.java:127]: @eeverman  Added tests. Please, check if they're ok, then I'll merge it.

**Review by eeverman — APPROVED** [review]
Looks good. The Builders are hard to test, but this does it. Thanks!
